### ISSUE TITLE
How to show different SEO description for a page with param

### ISSUE THREAD

**Issue by danfathom** [nystudio107/craft-seomatic#1323]
Hi,

I've got a page that has the URL of:
/testing

I also have a link that goes to the same page with an added query parameter for example:
/testing?content=myContent

I would like to have a different SEO description for both pages (I know technically they are the same page, but some content is different). Do you know how I could do this?

I've tried adding the following in the SEO description field for that page:
{% if craft.app.request.getParam('content') == 'myContent' %}test 1{% else %}test 2{% endif %}

This technically works, but because of the meta caching, it shows the description of the first page thats loaded. I can see in the plugins advances settings there is an option to change the meta cache duration, but the lowest option is 1 hour.

Do you have any suggestions of what I could do for this?

Thanks!

**Comment by khalwat** [comment]
You can do this by adding `content` to the `allowedUrlParams` setting: https://github.com/nystudio107/craft-seomatic/blob/develop-v4/src/config.php#L160

**Comment by danfathom** [comment]
Hi,

I've set this up on a different project we have, and i've got the seomatic.php file within the config folder with the following contents:

use nystudio107\seomatic\base\SeoElementInterface;

return [

   // string[] URL params that are allowed to be considered part of the unique URL used for the metadata cache
    'allowedUrlParams' => [
        'content'
    ]
    
];

I then have the following in one of my pages meta title source:

{% if craft.app.request.getParam('destination')|length %}Places to visit {{craft.app.request.getParam('destination')|replace('-', ' ')|capitalize}}{% else %}Places to visit{% endif %}

Strangely this works fine on my local, however on production it still caches. It's like SEOmatic isn't reading the seomatic.php on our production site. I've got the seomatic.php file in the config folder on both local and production.

It's probably something obvious i'm missing. I've cleared the SEOmatic meta cache multiple times and it updates on the first load but then caches again.

I don't suppose you have any ideas?

Many thanks,
Dan
